### PR TITLE
remove EXPERIMENTAL for tidb_partition_prune_mode = 'dynamic'

### PR DESCRIPTION
**Pull request by mjonss**
### First-time contributors' checklist 

- [X] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Making dynamic table partition prune mode General Available.

### Which TiDB version(s) do your changes apply to? (Required)
6.1

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [X] master (the latest development version)
- [X] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -503,19 +503,27 @@ To access partitioned tables in the MPP mode, you need to enable [dynamic prunin
 
 > **Warning:**
 >
-> - Currently, dynamic pruning mode for partitioned tables is an experimental feature and is not recommended for production environments.
->
-> - Do not enable dynamic pruning mode when a partitioned table contains columns of the `time` type. Otherwise, TiFlash crashes when a query selects a column of the `time` type.
+> - Do not enable dynamic pruning mode when a partitioned table contains columns of the `time` type. Otherwise, TiFlash crashes when a query selects a column of the `time` type. (TODO: Check if this is still correct?!)
```
Comment (shichun-0415) [inline tiflash/use-tiflash.md:506]:
```suggestion
```

**Comment by shichun-0415** [comment]
Hi, @mjonss, when you finish working on this PR. Please label it as ready for review. Thanks.

File: tiflash/use-tiflash.md (hunk 1 above)
Comment (mjonss) [reply tiflash/use-tiflash.md]: Check and did not find any error when doing a simple test. Removed.

**Review by Oreoxmt — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Review by moyun — COMMENTED** [review]
LGTM

File: partitioned-table.md [hunk 2]
```diff
@@ -1356,78 +1352,21 @@ mysql> explain select * from t1 where id < 150;
 
 From the above query results, you can see that the `Union` operator in the execution plan disappears while the partition pruning still takes effect and the execution plan only accesses `p0` and `p1`.
 
-`dynamic` mode makes execution plans simpler and clearer. Omitting the Union operation can improve the execution efficiency and avoid the problem of Union concurrent execution. In addition, `dynamic` mode also solves two problems that cannot be solved in `static` mode:
```
Comment (bb7133) [inline partitioned-table.md:1359]: Could we add some known limitations, for example, 

1) 'Prepared plan cache' cannot be applied with dynamic pruning mode.
2) Pruning with 'like ...' condition is not supported yet.

Comment (mjonss) [reply partitioned-table.md:1359]: @bb7133 could you take a new look? I added the pruning limitation under the "Partition pruning" section and "prepared statement plan cache" under the Dynamic prune mode.

File: partitioned-table.md [hunk 3]
```diff
@@ -724,6 +724,8 @@ The optimizer can prune partitions through `WHERE` conditions in the following t
 * partition_column = constant
 * partition_column IN (constant1, constant2, ..., constantN)
 
+It cannot currently prune partitions on LIKE conditions.
```
Comment (Oreoxmt) [inline partitioned-table.md:727]:
```suggestion
It cannot currently prune partitions on `LIKE` conditions.
```

File: partitioned-table.md [hunk 4]
```diff
@@ -1484,4 +1433,7 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
 8 rows in set (0.00 sec)
 ```
 
-From example 4, you can see that in `dynamic` mode, the execution plan with IndexJoin is selected when you execute the query.
+From example 2, you can see that in `dynamic` mode, the execution plan with IndexJoin is selected when you execute the query.
+
+
```
Comment (Oreoxmt) [inline partitioned-table.md:1438]:
```suggestion
```

File: partitioned-table.md [hunk 5]
```diff
@@ -1484,4 +1433,7 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
...
+Neither static or dynamic prune mode supports prepared statements plan cache. Dynamic mode may support it in a future version.
```
Comment (shichun-0415) [inline partitioned-table.md:1439]: Usually, we do not promise functions in future versions.
```suggestion
Currently, neither static nor dynamic pruning mode supports prepared statements plan cache. 
```

**Review by bb7133 — APPROVED** [review]
LGTM

File: partitioned-table.md [hunk 6]
```diff
@@ -724,6 +724,8 @@ The optimizer can prune partitions through `WHERE` conditions in the following t
 * partition_column = constant
 * partition_column IN (constant1, constant2, ..., constantN)
 
+It cannot currently prune partitions on `LIKE` conditions.
```
Comment (shichun-0415) [inline partitioned-table.md:727]:
```suggestion
Currently, partition pruning does not work with `LIKE` conditions.
```

**Pull request merged**